src/hotspot/share/runtime/continuation.cpp

```diff
@@ -160,6 +160,13 @@ freeze_result Continuation::try_preempt(JavaThread* target, oop continuation) {
   freeze_result res = CAST_TO_FN_PTR(FreezeContFnT, freeze_preempt_entry())(target, target->last_Java_sp());
   log_trace(continuations, preempt)("try_preempt: %d", res);
   JVMTI_ONLY(jubm.set_result(res);)
+
+  if (target->has_pending_exception()) {
+    assert(res == freeze_exception, "");
```

Comment:
Can you add an assert string as additional documentation, something like "expecting an exception result from freeze";

Reply:
Added.